### PR TITLE
Add print_return_invisible check

### LINKED ISSUE #49: Check if print method returns the object, invisibly
It is good practice to do this.

### PR DESCRIPTION
**Pull request by drmowinckels**
## Summary
- New `print_return_invisible` check that flags S3 `print.*` methods not returning `invisible(x)`
- Returning invisibly is standard practice for print methods and allows chaining

Closes #49

## Test plan
- [x] `bad_print` fixture with print method lacking `invisible()` triggers failure
- [x] `good` fixture passes (no print methods)
- [x] Edge case: no R directory passes
- [x] All existing tests still pass
- [x] Snapshot updated

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by mpadge — CHANGES_REQUESTED** [review]
See https://github.com/ropensci-review-tools/goodpractice/issues/221

File: R/chk_print_return.R [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    enc <- tryCatch(
+      desc::desc_get_field("Encoding", default = "UTF-8", file = path),
+      error = function(e) "UTF-8"
+    )
+
+    rfiles <- list.files(rdir, pattern = "\\.[rR]$", full.names = TRUE)
+    problems <- list()
+
+    for (f in rfiles) {
+      exprs <- tryCatch(
```
Comment (mpadge) [inline R/chk_print_return.R]: Dunno if that's your design or one from Claude, but I really like it 😄  Very clever!

Comment (drmowinckels) [reply R/chk_print_return.R]: I'm not gonna take credit for it. Just explained the issue and this was it's first solution 😊
